third-party/crowdsec/src/test/java/org/apache/james/CrowdsecEhloHookTest.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.james.protocols.smtp.utils.BaseFakeSMTPSession;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class CrowdsecEhloHookTest {
+    @RegisterExtension
+    static CrowdsecExtension crowdsecExtension = new CrowdsecExtension();
+
+    @BeforeAll
+    static void setUp() throws IOException, InterruptedException {
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli", "bouncer", "add", "bouncer", "-k", DEFAULT_API_KEY);
+    }
+
+    @Test
+    void givenIPBannedByCrowdsecDecisionIp() throws IOException, InterruptedException {
+        banIP("-i", "127.0.0.1");
+        int port = crowdsecExtension.getCrowdsecContainer().getMappedPort(CROWDSEC_PORT);
+        CrowdsecEhloHook ehloHook = new CrowdsecEhloHook(new CrowdsecClientConfiguration(new URL("http://localhost:" + port + "/v1"), CrowdsecClientConfiguration.DEFAULT_API_KEY));
+        SMTPSession session = new BaseFakeSMTPSession() {};
+
+         assertThat(ehloHook.doHelo(session, "localhost")).isEqualTo(HookResult.DENY);
+    }
+
+    @Test
+    void givenIPBannedByCrowdsecDecisionIpRange() throws IOException, InterruptedException {
+        banIP("-r", "127.0.0.1/24");
+        int port = crowdsecExtension.getCrowdsecContainer().getMappedPort(CROWDSEC_PORT);
+        CrowdsecEhloHook ehloHook = new CrowdsecEhloHook(new CrowdsecClientConfiguration(new URL("http://localhost:" + port + "/v1"), CrowdsecClientConfiguration.DEFAULT_API_KEY));
+        SMTPSession session = new BaseFakeSMTPSession() {};
+
+        assertThat(ehloHook.doHelo(session, "localhost")).isEqualTo(HookResult.DENY);
+    }
+
+    @Test
+    void givenIPNotBannedByCrowdsecDecisionIp() throws IOException, InterruptedException {
+        banIP("-i", "192.182.39.2");
+        int port = crowdsecExtension.getCrowdsecContainer().getMappedPort(CROWDSEC_PORT);
+        CrowdsecEhloHook ehloHook = new CrowdsecEhloHook(new CrowdsecClientConfiguration(new URL("http://localhost:" + port + "/v1"), CrowdsecClientConfiguration.DEFAULT_API_KEY));
+        SMTPSession session = new BaseFakeSMTPSession() {};
+
+        assertThat(ehloHook.doHelo(session, "localhost")).isEqualTo(HookResult.DECLINED);
+    }
+
+    @Test
+    void givenIPNotBannedByCrowdsecDecisionIpRange() throws IOException, InterruptedException {
+        banIP("-r", "192.182.39.2/24");
+        int port = crowdsecExtension.getCrowdsecContainer().getMappedPort(CROWDSEC_PORT);
+        CrowdsecEhloHook ehloHook = new CrowdsecEhloHook(new CrowdsecClientConfiguration(new URL("http://localhost:" + port + "/v1"), CrowdsecClientConfiguration.DEFAULT_API_KEY));
+        SMTPSession session = new BaseFakeSMTPSession() {};
+
+        assertThat(ehloHook.doHelo(session, "localhost")).isEqualTo(HookResult.DECLINED);
+    }
+
+    private static void banIP(String type, String value) throws IOException, InterruptedException {
```

Comment:
`-i` and `-r` do not speak at all to me.

I suggest expliciting this...

Does this support longer options? eg `--range` ?

Reply:
Yep, I will change it